### DISCUSSION TITLE
Windows : GAP Appearance read request timeout, irrespective of Attribute not found message from device

### DISCUSSION THREAD

**Discussion by sony-eldhose** [hbldh/bleak#1024] [Q&A]
* bleak version: 0.17.0
* Python version: 3.7.5
* Operating System: Windows 10
* BlueZ version (`bluetoothctl -v`) in case of Linux:

### Description

I am using bleak to connect to device . The GAP Read request for appearance is not handled in the device and response is not sent.  This is causing a timeout and the connection to fail. But the the device sent a response to Read by type request  for the same attribute  Appearance with a “Attribute not found “ response , windowsRT seems to ignore this and the read request for appearance is sent again.  The connection is successful in Mac OS with the same device , as it never sends a Read request for this attribute specifically .  
Please find the wireshark log 
![wireshark_log_gap_appearance_timeout](https://user-images.githubusercontent.com/104866608/191730936-4d3f890c-69d9-4965-9b55-1fa94f2dadd4.png)

**Comment by dlech** [comment]
The appearance characteristic is mandatory according to the Bluetooth spec (i.e. Table 12.2 in the v5.2 core specification). Some operating system Bluetooth stacks are more forgiving of non-complaint devices than other's. If you control the devices firmware, you should implement this characteristic (and any other mandatory ones).
